internal/dataplane/translator/subtranslator/ingress_atc_test.go

```diff
@@ -276,18 +281,141 @@ func TestTranslateIngressATC(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "KongServiceFacade used as a backend",
+			ingress: &netv1.Ingress{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-ingress",
+					Namespace: "default",
+				},
+				Spec: netv1.IngressSpec{
+					Rules: []netv1.IngressRule{{
+						Host: "konghq.com",
+						IngressRuleValue: netv1.IngressRuleValue{
+							HTTP: &netv1.HTTPIngressRuleValue{
+								Paths: []netv1.HTTPIngressPath{{
+									Path: "/api/",
+									Backend: netv1.IngressBackend{
+										Resource: &corev1.TypedLocalObjectReference{
+											APIGroup: lo.ToPtr(incubatorv1alpha1.GroupVersion.Group),
+											Kind:     incubatorv1alpha1.KongServiceFacadeKind,
+											Name:     "svc-facade",
+										},
+									},
+								}},
+							},
+						},
+					}},
+				},
+			},
+			kongServiceFacades: []*incubatorv1alpha1.KongServiceFacade{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "svc-facade",
+						Namespace: "default",
+					},
+					Spec: incubatorv1alpha1.KongServiceFacadeSpec{
+						Backend: incubatorv1alpha1.KongServiceFacadeBackend{
+							Name: "svc",
+							Port: 8080,
+						},
+					},
+				},
+			},
+			expectedServices: map[string]kongstate.Service{
+				"default.svc-facade.svc.facade": {
+					Namespace: corev1.NamespaceDefault,
+					Service: kong.Service{
+						Name:           kong.String("default.svc-facade.svc.facade"),
+						Host:           kong.String("default.svc-facade.svc.facade"),
+						ConnectTimeout: kong.Int(int(defaultServiceTimeout.Milliseconds())),
```

Comment:
This is more like food for thought to maybe address in a separate PR, in many places, we convert go `time.Duration` to let's call it Kong duration - milliseconds as *int. We should help for it or defaults even return with function in expected by Kong format. I can imagine making a silly mistake somewhere like calling `.Microseconds()`

Reply:
Good idea 👍 https://github.com/Kong/kubernetes-ingress-controller/pull/5262